### PR TITLE
Clean up if an exception is thrown in the middle of ServerFactory.Start

### PR DESCRIPTION
**Pull request by halter73**
#102

### CODE REVIEW

File: src/Microsoft.AspNet.Server.Kestrel/ServerFactory.cs [hunk 1]
```diff
@@ -36,30 +36,44 @@ public IFeatureCollection Initialize(IConfiguration configuration)
...
-                    address.Port,
-                    async frame =>
-                    {
-                        var request = new ServerRequest(frame);
-                        await application.Invoke(request.Features).ConfigureAwait(false);
-                    }));
-            }
-            disposables.Add(engine);
-            return new Disposable(() =>
+            var disposables = new Stack<IDisposable>();
```
Comment (Tratcher) [inline src/Microsoft.AspNet.Server.Kestrel/ServerFactory.cs]: Why a stack?

**Comment by Tratcher** [comment]
:shipit:

File: src/Microsoft.AspNet.Server.Kestrel/ServerFactory.cs (hunk 1 above)
Comment (halter73) [inline src/Microsoft.AspNet.Server.Kestrel/ServerFactory.cs]: Just to make sure the engine gets disposed last.

**Comment by lodejard** [comment]
:shipit:

**Pull request merged**